### PR TITLE
Add new TagPage component

### LINKED ISSUE #10824: Add TagPage component
spun off #10789

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #10824, spun off draft PR #10820

### CODE REVIEW

File: src/amo/pages/TagPage/index.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+} from 'amo/searchUtils';
+import type { SearchFilters } from 'amo/api/search';
+import type { AppState } from 'amo/store';
+import type { I18nType } from 'amo/types/i18n';
+import type { ReactRouterMatchType } from 'amo/types/router';
+
+type Props = {|
+  match: {|
+    ...ReactRouterMatchType,
+    params: {| tag: string, visibleAddonType: string |},
```
Comment (willdurand) [inline src/amo/pages/TagPage/index.js:24]: Is `visibleAddonType` a URL parameter? I thought the URL pattern was only `/tag/:tag/` and if that's the case, `visibleAddonType` should be removed here.

File: tests/unit/amo/pages/TestTagPage.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+import {
+  dispatchClientMetadata,
+  fakeI18n,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
+
+describe(__filename, () => {
+  let store;
+
+  const defaultParams = {
```
Comment (willdurand) [inline tests/unit/amo/pages/TestTagPage.js:19]: You can probably inline this variable in `render`:

```js
   function render({ params = { tag: 'some-tag' }, ...props } = {}) {
```

**Review by willdurand — APPROVED** [review]
r+wc, thanks

File: src/amo/pages/TagPage/index.js (hunk 1 above)
Comment (eviljeff) [reply src/amo/pages/TagPage/index.js]: https://github.com/mozilla/addons-frontend/pull/10825/commits/0922f516e66a0345d790d53e67f58ab1e6d03115

**Pull request merged**